tests/lib/tools/nested-state

```diff
@@ -0,0 +1,105 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: prepare"
+    echo "       restore"
+    echo "       build-core-image"
+    echo "       start-new-core-vm [--param-cd PARAM] [--param-mem PARAM]"
+    echo "       build-classic-image"
+    echo "       start-new-classic-vm"
+    echo "       start-existing-vm"
+    echo "       shutdown-existing-vm"
+    echo "       delete-existing-vm"
+    echo ""
+    echo "Available options:"
+    echo "  -h --help   show this help message."
+    echo ""
+}
+
+prepare() {
+    nested_prepare_env
+}
+
+restore() {
+    nested_cleanup_env
+}
+
+build_core_image() {
+    nested_create_core_vm
+}
+
+start_new_core_vm() {
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            --param-cd)
+                export NESTED_PARAM_CD="$2"
```

Comment:
Does it need to be exported? It seems only `nested.sh` actually uses it.

Reply:
We need to export that to then call the function in nested.sh which is using it

Reply:
Hmm, but exporting is only necessary for other processes, the shell should run this with the environment of the shell. Maybe I'm missing something.

Reply:
It is the only way to use the parameters. Then in a following PR I'll pass that as parameters to the functions instead of exporting the variables.